### PR TITLE
vim-patch:9.0.0012: signature files not detected properly

### PR DESCRIPTION
**Pull request by clason**
Problem:    Signature files not detected properly.
Solution:   Add a function to better detect signature files. (Doug Kearns)
https://github.com/vim/vim/commit/cdbfc6dbab1d63aa56af316d6b13e37939e7f7a8

### CODE REVIEW

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -847,6 +847,27 @@ func dist#ft#FTperl()
...
 endfunc
 
+" LambdaProlog and Standard ML signature files
+func dist#ft#FTsig()
+  if exists("g:filetype_sig")
+    exe "setf " .. g:filetype_sig
+    return
+  endif
+
+  let lprolog_comment = '^\s*\%(/\*\|%\)'
```
Comment (smjonas) [inline runtime/autoload/dist/ft.vim:857]: We try to use `findany` wherever possible, I'll send you an updated version in a few minutes :)

**Comment by clason** [comment]
These checks can probably be done with Lua patterns, but I'd need an adult to help with that :)

**Comment by smjonas** [comment]
I can send an updated version in a few minutes :)

**Comment by smjonas** [comment]
Here you go:
```lua
-- LambdaProlog and Standard ML signature files
function M.sig(bufnr)
  if vim.g.filetype_sig then
    return vim.g.filetype_sig
  end

  local line = nextnonblank(bufnr, 1)

  -- LambdaProlog comment or keyword
  if findany(line, { '^%s*/%*', '^%s*%%', '^%s*sig%s+%a' }) then
    return 'lprolog'
    -- SML comment or keyword
  elseif findany(line, { '^%s*%(%*', '^%s*signature%s+%a', '^%s*structure%s+%a' }) then
    return 'sml'
  end
end
```

**Comment by clason** [comment]
Much nicer, thank you very much!

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**